resolwe/flow/serializers/fields.py

```diff
@@ -72,4 +73,5 @@ def to_internal_value(self, data):
 
     def to_representation(self, obj):
         """Convert to representation."""
-        return self.serializer(obj, required=self.required).data
+        data = self.serializer(obj, required=self.required).data
+        return apply_subfield_projection(self, data, deep=True)
```

Comment:
This is not a good solution, because `.data` calculates all fields and only then applies projection.

Reply:
Why dont you use [`SelectiveFieldMixin`](https://github.com/genialis/resolwe/blob/7ee372a9700a4f3fedd621cf551567354741d9aa/resolwe/rest/serializers.py#L7)?

Reply:
We use the same approach in [ProjectableJSONField](https://github.com/genialis/resolwe/blob/master/resolwe/rest/fields.py#L12).

`SelectiveFieldMixin` is already in use through `ResolweBaseSerializer` but it does not suffice.

Reply:
Difference is that JSON field is read as a whole from the database and you get a dict. In this case you have to build that dict (serialize an object), so there is a lot overhead if you do it in a wrong order. Actually you lose all benefits gained by limiting the fields on the API.

Reply:
All nested object in DRF are [presented as serializers](https://www.django-rest-framework.org/api-guide/relations/#nested-relationships), not as fields. I think that this extra field around the actual serializer is causing problems.

Reply:
True, thanks for the hint. I believe i have now correctly addressed this.